### ISSUE TITLE
`uc_mem_protect` can't set readonly page to be writable

### ISSUE THREAD

**Issue by bet4it** [unicorn-engine/unicorn#1468]
```c
#include <unicorn/unicorn.h>

const unsigned char CODE[64] =  {0x01, 0x70, 0x04}; // add [eax+4], esi
unsigned char MEM[5] =  {0x00, 0x00, 0x00, 0x00};

int main()
{
    uc_engine *uc;
    int r_eax = 0x2000;
    int r_esi = 0xdeadbeef;
    uc_open(UC_ARCH_X86, UC_MODE_32, &uc);
    uc_reg_write(uc, UC_X86_REG_EAX, &r_eax);
    uc_reg_write(uc, UC_X86_REG_ESI, &r_esi);
    uc_mem_map(uc, 0x1000, 0x1000, UC_PROT_READ | UC_PROT_EXEC);
    uc_mem_write(uc, 0x1000, CODE, sizeof(CODE));
    uc_mem_map(uc, 0x2000, 0x1000, UC_PROT_READ);
    uc_mem_protect(uc, 0x2000, 0x1000, UC_PROT_READ | UC_PROT_WRITE);
    uc_emu_start(uc, 0x1000, 0x2000, 0, 1);
    uc_mem_read(uc, 0x2004, MEM, sizeof(MEM));
    printf("%02x %02x %02x %02x\n", MEM[0], MEM[1], MEM[2], MEM[3]);
    return 0;
}

```

**Comment by wtdcode** [comment]
Fixed in Qnicorn 1.0.0

**Comment by wtdcode** [comment]
Fixed in 2f61592ff9147363ea7f47e35226c218ebaf73bf
